code/modules/mining/equipment/monster_organs/monster_organ.dm

```diff
@@ -83,6 +86,15 @@
 	deltimer(decay_timer)
 	return ..()
 
+/obj/item/organ/monster_core/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if (layer_swapped)
+		return
+	// After being picked up, etc, reset our layer to that of a normal item
+	layer_swapped = TRUE
+	if (layer == initial(layer))
+		layer = /obj/item/organ::layer
```

Comment:
i dislike having to add extra work onto `Moved()` every single time as a result of this change. Can you make it so that this operates in its on proc as a `SIGNAL_HANDLER` and then once we get the signal once, we immediately deregister the signal? allows us to remove the boolean as well; feels like a much cleaner implementation.

Reply:
Overhead from just checking a var every time we move is essentially nonexistent. Signal self-registering is something to be avoided in general, so I believe this is cleaner.

Reply:
feels clunkier is all

Reply:
Disagree, see ping on discord with justification.
(Mirroring here jic) ``i'm not sure if overriding a proc is "clunkier" than self-registering a signal and then unregistering it once it triggers just so you don't have to varcheck once per said proc's call``

Reply:
I'd normally agree w san but self registration can be really dicey cause you end up with the same source/listener/signal different PROC problem, and idk if that's worth the cleanliness. would work as an element tho do we want to do this anywhere else?